src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,44 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final double MAX_SIZE_OF_UNITS = 10.0;
+    private static final int COUNT_OF_FIGURE = 5;
+    private static final double SIZE_OF_RADIUS = 10.0;
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int figureType = random.nextInt(COUNT_OF_FIGURE);
+        switch (figureType) {
+            case 0:
+                return new Square(colorSupplier.getRandomColor().toLowerCase(), random.nextDouble()
+                        * MAX_SIZE_OF_UNITS);
```

Comment:
Creating a figure, don't pass expressions in the constructor. Create separate variables and pass them on for better code readability. For example: double sideSize = random.nextDouble() * MAX_SIZE_OF_UNITS; and then pass sideSize to the constructor.